src/components/Settings/Speech/Speech.container.js

```diff
@@ -44,20 +45,46 @@ export class SpeechContainer extends Component {
   state = {
     selectedVoiceIndex: 0,
     isVoiceOpen: false,
-    anchorEl: undefined
+    anchorEl: undefined,
+    elevenLabsConnected: false,
+    elevenLabsValidating: false,
+    elevenLabsConnectionError: null
   };
 
-  async componentDidMount() {}
+  componentDidUpdate(prevProps) {
+    const { speech, getVoices } = this.props;
+
+    // Refresh voices when ElevenLabs API key changes
+    if (prevProps.speech.elevenLabsApiKey !== speech.elevenLabsApiKey) {
+      getVoices();
+    }
+  }
 
   handleUpdateElevenLabsApiKey = async apiKey => {
     const { changeElevenLabsApiKey } = this.props;
 
-    if (apiKey && !validateApiKeyFormat(apiKey)) {
-      throw new Error('Invalid API key format');
-    }
```

Comment:
We should conserve a apiKey validation here. You can add an early return instead of throw an Error. 

Also you can set 

```
this.setState({
      elevenLabsConnected: result.isValid,
      elevenLabsValidating: false,
      elevenLabsConnectionError: result.isValid ? null : result.error
    });
```